File: JAVA/160. Intersection of Two Linked Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode A, ListNode B) {
        ListNode x=A;
        ListNode y=B;
        while(x!=y){
            x=x!=null?x.next:B;
            y=y!=null?y.next:A;
        }
        return x;
    }
}

File: C++/160. Intersection of Two Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *A, ListNode *B) {
        ListNode *x=A;
        ListNode *y=B;
        while(x!=y){
            x=x?x->next:A;
            y=y?y->next:B;
        }
        return x;
    }
};